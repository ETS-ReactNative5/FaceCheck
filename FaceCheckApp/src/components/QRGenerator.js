import firebase from 'react-native-firebase';
import React, {Component} from 'react';
import {View} from 'react-native';
import QRCode from 'react-native-qrcode-svg';
import {Surface} from 'react-native-paper';
import {hook, wrap} from 'cavy';

import styles from 'FaceCheckApp/src/assets/styles';

class QRGenerator extends Component {
  constructor(props) {
    super(props);
    this.state = {init: true, seconds: 0, uid: null};
  }

  componentDidMount() {
    this.interval = setInterval(() => {
      this.tick();
    }, 1000);
    const {uid} = firebase.auth().currentUser;
    this.setState({uid});
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  tick() {
    this.setState({seconds: new Date().getSeconds()});
  }

  shouldComponentUpdate() {
    return (this.state.seconds + 1) % 30 == 0;
  }

  render() {
    const qrData = JSON.stringify({
      uid: this.state.uid,
      totpCode: Math.floor(Math.random() * Math.floor(999999)),
    });
    const TestableFunctionComponent = wrap(QRCode);
    return (
      <View style={styles.centerItem}>
        <Surface style={styles.surface}>
          <TestableFunctionComponent
            ref={this.props.generateTestHook('Scene.QRCode')}
            style={{alignItems: 'center'}}
            value={qrData}
            size={150}
            ecl="Q"
          />
        </Surface>
      </View>
    );
  }
}

export default hook(QRGenerator);