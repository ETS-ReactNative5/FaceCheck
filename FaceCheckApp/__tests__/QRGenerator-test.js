import 'react-native';
import React from 'react';
import QRGenerator from '../src/screens/QRGeneratorScreen';
import RNFirebase from 'react-native-firebase';
import renderer from 'react-test-renderer';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import QRCode from 'react-native-qrcode-svg';

configure({ adapter: new Adapter() });

test('Test QR snapshot',()=>{
  let QRGeneratorData = renderer.create(<QRGenerator/>).toJSON;
  expect(QRGeneratorData).toMatchSnapshot();
})

test('Renders QR code', () => {
  const qrData = JSON.stringify({
    uid: 123,
    totpCode: Math.floor(Math.random() * Math.floor(999999)),
  });
  const wrapper = shallow(<QRCode value={qrData}/>);
  
  expect(wrapper.find('.value')).toBeDefined();
})

test('asymmetric matcher', () => {
  const expected = {a: expect.any(Function)};
  const received = {a: _in => _in};
  expect(received).toEqual(expected);
})
