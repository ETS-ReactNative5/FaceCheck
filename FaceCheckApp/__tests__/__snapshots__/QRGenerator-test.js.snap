// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test QR snapshot 1`] = `[Function]`;

exports[`check qr screen 1`] = `[Function]`;
